Add visual inaccuracy to effect warhead and use it on light RA/TD weapons

reaperrr opened the pull request:
Purely cosmetic, allows to 'fake' slight inaccuracy without affecting game balance on weapons like light machine guns.

Used on RA Minigunner M1Carbine and the M60 MGs in both RA and TD to drop the somewhat overkill `piffs` animation if favor of just `piff`.

reaperrr committed:
Support 'inaccuracy' on effect warhead

Purely cosmetic, allows to 'fake' slight inaccuracy without
affecting game balance on weapons like light machine guns.

reaperrr committed:
Move around TD Pistol yaml

Feels more organized if located below the inherited
^LightMG, like all other inheritors.

reaperrr committed:
Update RA/TD light guns cosmetic effects

Use single piff instead of piffs per impact on
M60s and M1, but in return add visual inaccuracy.

Gave minigunner MG two additional delayed effect warheads
to match sound and fire animation.

pchote commented:
The jeep/humvee look good, with multiple single piffs per burst, but IMO it looks weird for the minigunners to only generates a single piff total per burst. Can we change this?

reaperrr commented:
Updated. Minigunners' guns get 2 delayed piffs now to match sound and fire anim, looks indeed much better (than bleed, too).

pchote commented:
Looks really good now.

I found a couple of additional piff issues while testing, can we fix these while we are here?
* TD Commando doesn't show any piffs (should be a single piff)
* RA Spy shows heavy piffs (should be a single piff)
* RA Sniper doesn't show any piffs (should be a single piff)

reaperrr committed:
Fix piff anims of Sniper and SilencedPPK weapons

reaperrr committed:
Fix some RA effect warheads ignoring water actors

reaperrr commented:
I stumbled over those as well, but wasn't sure whether they may have been intentional at some point.
Anyway, fixed.

Also noticed a few remaining wrong `ValidTargets` setups that forgot to account for water actors. Fixed that as well.

abcdefg30 commented on OpenRA.Mods.Common/Warheads/CreateEffectWarhead.cs:
```diff
@@ -109,6 +112,9 @@ public override void DoImpact(Target target, WarheadArgs args)
 			if (Image != null && explosion != null)
 			{
+				if (Inaccuracy.Length > 0)
+					pos += WVec.FromPDF(world.SharedRandom, 2) * Inaccuracy.Length / 1024;
```
Why isn't this using the same logic we use in all other places for inaccuracy?

reaperrr replied:
I got this from `Bullet`, though...?

abcdefg30 replied:
This bullet? :P https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Common/Projectiles/Bullet.cs#L151-L155

reaperrr replied:
Implementing the other inaccuracy types on a warhead would require keeping track of the distance between source and target (including range modifiers and such), which opens a can of worms in terms of what args need to be added/passed from `ProjectileArgs` on to `WarheadArgs`.
It's certainly worth thinking about, but I don't feel like dealing with that in the short-term (iow, before next release).

pchote commented:
This is nice polish that fits with the other warhead changes, so adding to the milestone.

abcdefg30 reviewed (approved):
Although the remasters do it a bit better than we Imho.

abcdefg30 commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/255a36ff115656782e4d4df7c8a3128a7825d92b...ca9f98fb4aea6e13cc981f17716e3a1f9d0abda3)
